src/backend/distributed/operations/shard_split.c

```diff
@@ -1346,107 +1339,123 @@ NonBlockingShardSplit(SplitOperation splitOperation,
 		 * Subscriber flow starts from here.
 		 * Populate 'ShardSplitSubscriberMetadata' for subscription management.
 		 */
-		List *shardSplitSubscribersMetadataList =
+		List *subscriptionInfoList =
 			PopulateShardSplitSubscriptionsMetadataList(
-				shardSplitHashMapForPublication, replicationSlotInfoList);
+				publicationInfoHash, replicationSlotInfoList,
+				shardGroupSplitIntervalListList, workersForPlacementList);
+
+		HTAB *nodeSubscriptionsHash = CreateNodeSubscriptionsHash(subscriptionInfoList);
 
 		/* Create connections to the target nodes */
-		List *targetNodeConnectionList = CreateTargetNodeConnectionsForShardSplit(
-			shardSplitSubscribersMetadataList,
-			connectionFlags,
+		CreateNodeSubscriptionsConnections(
+			nodeSubscriptionsHash,
 			superUser, databaseName);
 
-		/* 7) Create copies of template replication slot */
-		char *templateSlotName = ShardSplitTemplateReplicationSlotName(
-			shardIntervalToSplit->shardId);
-		CreateReplicationSlots(sourceConnection, templateSlotName,
-							   shardSplitSubscribersMetadataList);
 
-		/* 8) Create subscriptions on target nodes */
-		CreateShardSplitSubscriptions(targetNodeConnectionList,
-									  shardSplitSubscribersMetadataList,
-									  sourceShardToCopyNode,
-									  superUser,
-									  databaseName);
+		/*
+		 * 6) Create replication slots and keep track of their snapshot.
+		 */
+		char *snapshot = CreateReplicationSlots(
+			sourceConnection,
+			sourceReplicationConnection,
+			subscriptionInfoList,
+			"citus");
```

Comment:
nit: Would be better to store in local variable and then pass.
char *decoderPlugin = "citus"